Fixes #38382 - TableIndexPage improvements

jeremylenz opened the pull request:
Goes with https://github.com/Katello/katello/pull/11364

To use Patternfly expandable table rows, each Tr must be enclosed in a Tbody. The `childrenOutsideTbody` prop of `<Table>` tried to address this, but there are a couple more changes needed.

1. When `childrenOutsideTbody` is passed, indicating use of expandable table rows, also render another column header for the expansion column. This way the column names do not get shifted incorrectly.
2. Pass the `isExpandable` prop to PFTable. This will tell Patternfly to properly render shading of table rows.

~~Also, this PR fixes some issues with pagination by allowing you to control both top and bottom pagination from outside TableIndexPage.~~

MariaAga commented:
> Also, this PR fixes some issues with pagination by allowing you to control both top and bottom pagination from outside TableIndexPage.

What issues? Why can't they be resolved in the tableindexpage?

MariaAga commented on webpack/assets/javascripts/react_app/components/PF4/TableIndexPage/Table/Table.js:
```diff
@@ -113,12 +113,20 @@ export const Table = ({
             {showCheckboxes && (
               <Th aria-label="checkbox-header" key="checkbox-th" />
             )}
+            {childrenOutsideTbody && (
+              <Th aria-label="expansion-carat" key="expansion-carat" />
+            )}
```
If we use this, instead of the column adjustment that I suggested in Katello, please open a PR in REX to remove remove the extra Th from there

jeremylenz replied:
Will do :+1:

MariaAga replied:
ping

jeremylenz replied:
Sorry for the delay :) https://github.com/theforeman/foreman_remote_execution/pull/972

jeremylenz commented:
> What issues? Why can't they be resolved in the tableindexpage?

Normally TableIndexPage renders the top pagination and Table renders the bottom pagination. That means if I pass children to TableIndexPage, I can pass in my own Pagination component for the bottom but not the top. IIRC it was causing the top pagination < > buttons to update the bottom pagination, but the bottom < > buttons were not updating the top.

MariaAga commented:
> Normally TableIndexPage renders the top pagination and Table renders the bottom pagination. That means if I pass children to TableIndexPage, I can pass in my own Pagination component for the bottom but not the top. IIRC it was causing the top pagination < > buttons to update the bottom pagination, but the bottom < > buttons were not updating the top.

From what I saw the issues were caused by sending over the wrong props, I sent comments in the Katello PR on how to fix them and it got both paginations working

jeremylenz committed:
Fixes #38382 - TableIndexPage improvements
      Add full support for expandable table rows
      Allow control of both top and bottom pagination component in TableIndexPage

jeremylenz commented:
@MariaAga Your fixes seem to work great, thanks! Updated

MariaAga commented:
Can I assume Katello and rpm failures are unrelated?

jeremylenz commented:
Looks like everything passed now, I won't worry about packit :)

MariaAga reviewed (approved):
Thanks!
